### DISCUSSION TITLE
React hook form and TinyMCE for React

### DISCUSSION THREAD

**Discussion by shestakov-vladyslav** [react-hook-form/react-hook-form#9049] [Q&A] [answered]
### Version Number

7.35.0

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-usefieldarray-nested-arrays-forked-k1urds?file=/src/nestedItem.js

### Steps to reproduce

Help me please 🤯

Go to https://codesandbox.io/s/react-hook-form-usefieldarray-nested-arrays-forked-k1urds?file=/src/nestedItem.js

1. Click "Append"
2. Click "Append nested"
3. Type something after "field1"
4. Unfocus field
4. E.R.: Works fine
A.R.: It rolls back to "field1" 🤯

![React Hook Form - useFieldArray nested arrays (forked) - CodeSandbox](https://user-images.githubusercontent.com/18069756/190680184-735fe3c2-a2c2-4362-8a0f-d338b854a63b.gif)

![Cyper](https://user-images.githubusercontent.com/18069756/190681776-29d22342-cd1f-439d-aa69-9167f6c15c8e.gif)

Please help me to figure what is wrong

P.S. It seems it can be connected to this issue https://github.com/react-hook-form/react-hook-form/issues/4857

### Expected behaviour

Nested controlled field value doesn't roll back on blur

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari, Edge

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by leapful** [comment, marked as answer]
@shestakov-vladyslav 

You are passing the whole "field" object of react-hook-form to `<Editor>` including "onChange" callback. The prop `<Editor onChange>` is also a valid prop for Editor component therefore it will be registered with "onChange" callback of react-hook-form but the real value change is only applied on "onEditorCallback" therefore Editor's value is reset to "undefined" when "onChange" is triggered.

Exclude "onChange" when passing to `<Editor>` and only call it inside "onEditorChange" will fix the issue.

Working demo: https://codesandbox.io/s/react-hook-form-usefieldarray-nested-arrays-forked-3ignce?file=/src/nestedItem.js

**Comment by shestakov-vladyslav** [comment]
You're my life-savior
Thanks for the support & help

**Comment by shestakov-vladyslav** [comment]
Also my second problem was caused by using useWatch for field arrays...
I just removed useWatch and used "fields" property instead and it works like a charm

Example of **wrong** using:
![image](https://user-images.githubusercontent.com/18069756/191010754-1b09e80b-0aab-4e4f-a4d9-abfed4b388e6.png)

And it happens only if you use useWatch for root useFieldArray

Wrong demo: https://codesandbox.io/s/react-hook-form-usefieldarray-nested-arrays-forked-7vfr8s?file=/src/nestedFieldArray.js:345-352

But maybe it's react hook form bug, or I just don't understand something

P.S. After all I replaced useWatch with watch and it works. Good tip to know, but hard to understand what happens underneath
